test/test_rtlog.cpp

```diff
@@ -66,12 +66,45 @@ static auto PrintMessage = [](const ExampleLogData &data, size_t sequenceNumber,
 
 using namespace rtlog::test;
 
+using SingleWriterRtLoggerType =
+    rtlog::Logger<ExampleLogData, MAX_NUM_LOG_MESSAGES, MAX_LOG_MESSAGE_LENGTH,
+                  gSequenceNumber, rtlog::SingleRealtimeWriterQueueType>;
+using MultiWriterRtLoggerType =
+    rtlog::Logger<ExampleLogData, MAX_NUM_LOG_MESSAGES, MAX_LOG_MESSAGE_LENGTH,
+                  gSequenceNumber, rtlog::MultiRealtimeWriterQueueType>;
+
+template <typename LoggerType> class RtLogTest : public ::testing::Test {
+protected:
+  LoggerType logger;
+};
+
+typedef ::testing::Types<SingleWriterRtLoggerType, MultiWriterRtLoggerType>
+    LoggerTypes;
+TYPED_TEST_SUITE(RtLogTest, LoggerTypes);
+
+using TruncatedSingleWriterRtLoggerType =
+    rtlog::Logger<ExampleLogData, MAX_NUM_LOG_MESSAGES, 10, gSequenceNumber,
+                  rtlog::SingleRealtimeWriterQueueType>;
+using TruncatedMultiWriterRtLoggerType =
+    rtlog::Logger<ExampleLogData, MAX_NUM_LOG_MESSAGES, 10, gSequenceNumber,
+                  rtlog::MultiRealtimeWriterQueueType>;
+
+template <typename LoggerType>
+class TruncatedRtLogTest : public ::testing::Test {
+protected:
+  LoggerType logger;
+  inline static const size_t maxMessageLength = 10;
+};
+
+typedef ::testing::Types<TruncatedSingleWriterRtLoggerType,
+                         TruncatedMultiWriterRtLoggerType>
+    TruncatedLoggerTypes;
+TYPED_TEST_SUITE(TruncatedRtLogTest, TruncatedLoggerTypes);
+
 #ifdef RTLOG_USE_STB
 
-TEST(RtlogTest, BasicConstruction) {
-  rtlog::Logger<ExampleLogData, MAX_NUM_LOG_MESSAGES, MAX_LOG_MESSAGE_LENGTH,
-                gSequenceNumber>
-      logger;
+TYPED_TEST(RtLogTest, BasicConstruction) {
+  auto &logger = this->logger;
```

Comment:
TODO: for future me, follow up with a review to change these out. Added these helpers to make the review more legible